packages/contracts/src/diamond/upgradeInitializers/DiamondInit.sol

```diff
@@ -0,0 +1,111 @@
+import {UbiquityGovernanceTokenForDiamond} from "../token/UbiquityGovernanceTokenForDiamond.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {LibTWAPOracle} from "../libraries/LibTWAPOracle.sol";
+import {LibDollar} from "../libraries/LibDollar.sol";
+import {LibStaking} from "../libraries/LibStaking.sol";
+import {LibChef} from "../libraries/LibChef.sol";
+import {LibCreditNftManager} from "../libraries/LibCreditNFTManager.sol";
+import {LibCreditRedemptionCalculator} from "../libraries/LibCreditRedemptionCalculator.sol";
+import {UbiquityDollarManager} from "../../dollar/core/UbiquityDollarManager.sol";
+
+// It is expected that this contract is customized if you want to deploy your diamond
+// with data from a deployment script. Use the init function to initialize state variables
+// of your diamond. Add parameters to the init function if you need to.
+
+contract DiamondInit is Modifiers {
+    struct Args {
+        address admin;
+        string dollarName;
+        string dollarSymbol;
+        uint8 dollarDecimals;
+        address[] tos;
+        uint256[] amounts;
+        uint256[] stakingShareIDs;
+        uint256 governancePerBlock;
+        uint256 creditNFTLengthBlocks;
+    }
+
+    // You can add parameters to this function in order to pass in
+    // data to set your own state variables
+    function init(Args memory _args) external {
+        // adding ERC165 data
+        LibDiamond.DiamondStorage storage ds = LibDiamond.diamondStorage();
+        ds.supportedInterfaces[type(IERC165).interfaceId] = true;
+        ds.supportedInterfaces[type(IDiamondCut).interfaceId] = true;
+        ds.supportedInterfaces[type(IDiamondLoupe).interfaceId] = true;
+        ds.supportedInterfaces[type(IERC173).interfaceId] = true;
+
+        LibAccessControl.grantRole(DEFAULT_ADMIN_ROLE, _args.admin);
+        LibAccessControl.grantRole(GOVERNANCE_TOKEN_MINTER_ROLE, _args.admin);
+        LibAccessControl.grantRole(PAUSER_ROLE, _args.admin);
+        LibAccessControl.grantRole(CREDIT_NFT_MANAGER_ROLE, _args.admin);
+        LibAccessControl.grantRole(STAKING_MANAGER_ROLE, _args.admin);
+        LibAccessControl.grantRole(INCENTIVE_MANAGER_ROLE, _args.admin);
+        LibAccessControl.grantRole(GOVERNANCE_TOKEN_MANAGER_ROLE, _args.admin);
+        AppStorage storage appStore = LibAppStorage.appStorage();
+
+        appStore.paused = false;
+        appStore.treasuryAddress = _args.admin;
+        // Dollar
+        LibDollar.initialize(
+            _args.dollarName,
+            _args.dollarSymbol,
+            _args.dollarDecimals
+        );
+        // staking
+        LibStaking.StakingData storage ls = LibStaking.stakingStorage();
+        ls.stakingDiscountMultiplier = uint256(0.001 ether); // 0.001
+        ls.blockCountInAWeek = 45361;
+        // add staking shares
+
+        string
+            memory uri = "https://bafybeifibz4fhk4yag5reupmgh5cdbm2oladke4zfd7ldyw7avgipocpmy.ipfs.infura-ipfs.io/";
```

Comment:
Seems like a bad idea to hardcode this in. We've had issues with it in the past. At least there is a setter right?

Reply:
yep it is used inside stakingshare which inherits ERC1155SetUri which provide an internal _setURI however stakingShare does not include an external function to modify setURI should we open an issue to add that feature ?

Reply:
https://github.com/ubiquity/ubiquity-dollar/issues/561